src/redux/data.ts

```diff
@@ -1492,105 +1512,102 @@ export const dataWatchPendingTransactions = (
+) => {
+  const { pendingTransactions: pending } = getState().data;
+  if (isEmpty(pending)) {
+    return true;
+  }
+  let txStatusesDidChange = false;
+  const updatedPendingTransactions = await Promise.all(
+    pending.map(async tx => {
+      const updatedPending = { ...tx };
+      const txHash = ethereumUtils.getHash(tx);
+      try {
+        logger.log('Checking pending tx with hash', txHash);
+        const p =
+          provider || (await getProviderForNetwork(updatedPending.network));
+        const txObj = await p.getTransaction(txHash!);
+        // if the nonce of last confirmed tx is higher than this pending tx then it got dropped
+        const nonceAlreadyIncluded = currentNonce > tx.nonce!;
+        if ((txObj?.blockNumber && txObj?.blockHash) || nonceAlreadyIncluded) {
+          // When speeding up a non "normal tx" we need to resubscribe
+          // because zerion "append" event isn't reliable
+          logger.log('TX CONFIRMED!', txObj);
+          if (!nonceAlreadyIncluded) {
+            appEvents.emit('transactionConfirmed', txObj);
+          }
+          const minedAt = Math.floor(Date.now() / 1000);
+          txStatusesDidChange = true;
+          // @ts-expect-error `txObj` is not typed as having a `status` field.
+          if (txObj && !isZero(txObj.status)) {
+            const isSelf = toLower(tx?.from!) === toLower(tx?.to!);
+            const newStatus = getTransactionLabel({
+              direction: isSelf
+                ? TransactionDirection.self
+                : TransactionDirection.out,
+              pending: false,
+              protocol: tx?.protocol,
+              status:
+                tx.status === TransactionStatus.cancelling
+                  ? TransactionStatus.cancelled
+                  : getConfirmedState(tx.type),
+              type: tx?.type,
             });
-            updatedPending.title = title;
-            updatedPending.pending = false;
-            updatedPending.minedAt = minedAt;
+            updatedPending.status = newStatus;
+          } else if (nonceAlreadyIncluded) {
+            updatedPending.status = TransactionStatus.unknown;
+          } else {
+            updatedPending.status = TransactionStatus.failed;
           }
-        } catch (error) {
-          logger.log('Error watching pending txn', error);
+          const title = getTitle({
+            protocol: tx.protocol,
+            status: updatedPending.status,
+            type: tx.type,
+          });
+          updatedPending.title = title;
+          updatedPending.pending = false;
+          updatedPending.minedAt = minedAt;
         }
-        return updatedPending;
-      })
+      } catch (error) {
+        logger.log('Error watching pending txn', error);
+      }
+      return updatedPending;
+    })
+  );
+
+  if (txStatusesDidChange) {
+    const { accountAddress, network } = getState().settings;
+    const [newDataTransactions, pendingTransactions] = partition(
```

Comment:
Making a note for discussion:
Previously, we would completely filter out anything that is of TransactionStatus.unknown
Now, we do a partition and keep status unknown in pendingTransactions?